api/api.go

```diff
@@ -0,0 +1,162 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/mp-hl-2021/chat/usecases"
+	"net/http"
+)
+
+
+
+type Api struct {
+	AccountUseCases usecases.AccountUseCasesInterface
+	LinkUseCases usecases.LinkUseCasesInterface
+}
+
+func NewApi(a usecases.AccountUseCasesInterface, l usecases.LinkUseCasesInterface) *Api {
+	return &Api{
+		AccountUseCases: a,
+		LinkUseCases: l,
+	}
+}
+
+// NewRouter creates all endpoint for chat app.
+func (a *Api) Router() http.Handler {
+	router := mux.NewRouter()
+
+
+
+	// / post request to create link <link to source>, returns <short link>
+	router.HandleFunc("/", a.postCreateLink).Methods(http.MethodPost)
```

Comment:
Don't you think it'll be more RESTful to use some explicit resource naming here ("/links"), instead of just a root endpoint. Please provide arguments to support your design choice.

Reply:
That's crucial design decision, which shouldn't be changed, as for me. Creating short link is a main purpose of our service, so it's reasonable that "root" path is the path to "root" function. That's probably not so RESTful, but ideologically important. For example, when you type "yandex.ru", you get main page of the yandex. You almost never type "yandex.ru/search" or "yandex.ru/engine" in order to access search engine. Also, If we use explicit naming, what will be placed by root path? 404? Redirect to other path? :polish_cow:

Reply:
OK, point taken.